keps/sig-node/1985-remove-dockershim/README.md

```diff
@@ -0,0 +1,298 @@
+needs to be another KEP for graduating the CRI API version.
+- Existing CNI and CSI plugins may also be affected.
+- cri-dockerd will vendor kubernetes/kubernetes, that may be tough.
+- cri-dockerd as an independent software running on node should be allocated enough resource to guarantee its availability.
+
+> You can check [the discussion in sig-node mailing list](https://groups.google.com/forum/#!msg/kubernetes-sig-node/0qVzfugYhro/l6Au216XAgAJ) for more details. 
+
+### Goals
+
+- A concrete dockershim removal criteria.
+- A brief plan to remove dockershim spanning multiple releases.
+
+### Non-Goals
+
+- Refactoring or re-design of dockershim itself due to deprecation.
+
+## Proposal
+
+### Dockershim removal criteria
+
+- CRI itself is alpha. So we need another KEP to graduate CRI API.
+- kubelet has no dependency on dockershim/docker in its whole lifecycle. This is already done using the `dockerless` tag
+- All node related features are CRI generic and have no "back door" dependency on dockershim/docker.
+- Deprecate and remove, or replace all Docker-specific features.
+- Reasonable benchmark result of performance degradation after moving dockershim to out-of-tree.
+- E2E test framework has been updated with fully support of out-of-tree CRI container runtime.
+
+### Dockershim removal plan
+
+Step 1: Deprecate in-tree dockershim and decouple dockershim from kubelet.
+
+Target releases: 1.20, 1.21
+
+Actions:
+
+- Mark in-tree dockershim as "maintenance mode":
+  - CRI generic changes/features can continue on dockershim.
+  - WIP efforts on dockershim can continue and go to complete.
+  - dockershim/docker specific changes/features should be rejected.
+- Deprecate the legacy features of dockershim in kubelet by providing a specific timeline. Currently, kubelet still has:
+  - vendored dockershim 
+  - flags that are used to configure dockershim.
+  - support to get container logs when docker uses journald as the driver.
+  - logic of moving docker processes to a given cgroup
+  - TBD anything else?
+- Ensure e2e/Node e2e test framework is CRI generic and test cases are independent of container runtime.
+- Refactoring e2e/Node e2e test framework to include CRI for docker installation (or use other CRI container runtime).
+  - Ensure cluster/node e2e are 100% CRI focused.
+  - Ensure test-infra install appropriate CRI implementations in e2e machines. 
+- Ensure Windows scenarios that currently depend on docker are fully supported in alternative CRI implementations
+
+Step 2: Release kubelet without dockershim
+
+Target releases: 1.22
+
+Actions:
+- Document and announce migration guide.
+- Release harness would build kubelet with `dockerless` tag on. So the default build will not support docker out of 
+the box. 
+- If folks need this support, they would have to build kublet by themselves as the code is still present in the
+source tree.
```

Comment:
at first glance, this doesn't match the spirit of the deprecation policy, which is intended to avoid requiring consumers to change the way they consume/run kubernetes until the deprecation period ends

Reply:
Let's tweak this as we go forward as the KEP is a living document.